backend/pkg/notification/notifications.go

```diff
@@ -2317,53 +2065,55 @@ func collectMonitoringMachineMemoryUsage(notificationsByUserID map[uint64]map[ty
 var isFirstNotificationCheck = true
 
 func collectMonitoringMachine(
-	notificationsByUserID map[uint64]map[types.EventName][]types.Notification,
+	notificationsByUserID types.NotificationsPerUserId,
 	eventName types.EventName,
 	epochWaitInBetween int,
-	notifyConditionFulfilled func(subscribeData *MachineEvents, machineData *types.MachineMetricSystemUser) bool,
+	notifyConditionFulfilled func(subscribeData *types.Subscription, machineData *types.MachineMetricSystemUser) bool,
 	epoch uint64,
 ) error {
-	var allSubscribed []MachineEvents
+	var allSubscribed []*types.Subscription
+	// event_filter == machine name
+	// TODO: clarify why we need grouping here?!
```

Comment:
Good question, I don't think we need it either. Ran a couple queries against the db to see if I can spot an edge case but result set of grouped and ungrouped is the same for all machine type notifications.

Reply:
removed the grouping and switched func to use the `GetSubsForEventFilter` function